src/Jinja2CppLight.h

```diff
@@ -213,8 +214,8 @@ class ForSection : public ControlSection {
         if( valueByName.find( varName ) != valueByName.end() ) {
             throw render_error("variable " + varName + " already exists in this context" );
         }
-        for( auto &v : values->values ) {
-            valueByName[varName] = v;
+        for (auto i = loopStart; i < loopEnd; ++i ){
+            valueByName[varName] = std::make_shared<IntValue>( i );
```

Comment:
May be it's a good idea to make shared pointer to IntValule once (outside the loop). Something like this:
```c++
auto intVal = std::make_shared<IntValue>(0);
valueByName[varName] = intVal;
for (auto i = loopStart; i < loopEnd; ++i) {
     *intVal = IntValue(i);
```

And don't forget to clear 'varName' slot after the rendering (same as in line 257)